python/cudf/cudf/core/series.py

```diff
@@ -1455,11 +1455,12 @@ def astype(self, dtype, copy=True, errors="raise", **kwargs):
                 )
             dtype = dtype[self.name]
 
+        result = self.copy(deep=copy)
```

Comment:
This will deep copy the column which then may end up being copied again if the dtype can't be changed zero copy.